### PR TITLE
Add more unit test cases for EditableText widget

### PR DESCRIPTION
**Pull request by xu-baolin**
## Description

Add regression test for #66036 

The root cause of the issue is that the keyboard was requested incorrectly after receiving repeated updated values.
I added some test cases to test this situation, and also tested the scenario where the keyboard is hidden.

## Related Issues

#66036 

## Tests

See files.

### CODE REVIEW

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -4998,6 +4998,86 @@ void main() {
...
+    expect(focusNode.hasFocus, true);
+
+    log.clear();
+
+    final EditableTextState state = tester.firstState(find.byType(EditableText));
+
+    state.updateEditingValue(const TextEditingValue(
+      text: 'a',
+    ));
+    // Nothing called when only the remote changes
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5052]: This needs a period at the end.

Comment (justinmc) [reply packages/flutter/test/widgets/editable_text_test.dart:5052]: Same for a few other comments below.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -4998,6 +4998,86 @@ void main() {
...
+    // Close the InputConnection.
+    expect(methodCall, isMethodCall('TextInput.clearClient', arguments: null),);
+    methodCall = log[1];
+    expect(methodCall, isMethodCall('TextInput.hide', arguments: null),);
+    // The keyboard lose focus.
+    expect(focusNode.hasFocus, false);
+
+    log.clear();
+
+    // The keyboard does not be requested when receiving repeat value from the engine
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5069]: "does not be" => "is not"

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 3]
```diff
@@ -4998,6 +4998,86 @@ void main() {
...
+    state.updateEditingValue(const TextEditingValue(
+      text: 'a',
+    ));
+    // Nothing called when only the remote changes
+    expect(log.length, 0);
+    await tester.pump();
+
+    //The keyboard does not be requested after a repeat value from the engine.
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5077]: Another "does not be" => "is not"

Also add a space at the start: "// The"

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 4]
```diff
@@ -4998,6 +4998,86 @@ void main() {
...
+    // Hide the keyboard
+    focusNode.unfocus();
+    await tester.pump();
+    expect(log.length, 2);
+    MethodCall methodCall = log[0];
+    // Close the InputConnection.
+    expect(methodCall, isMethodCall('TextInput.clearClient', arguments: null),);
+    methodCall = log[1];
+    expect(methodCall, isMethodCall('TextInput.hide', arguments: null),);
+    // The keyboard lose focus.
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5064]: "lose" => "loses"

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 5]
```diff
@@ -4998,6 +4998,86 @@ void main() {
     );
   });
 
+  testWidgets('Receive repeat [TextEditingValue] will not trigger the request keyboard', (WidgetTester tester) async {
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5001]: "Repeatedly receiving [TextEditingValue] will not trigger a keyboard request"

**Review by justinmc — COMMENTED** [review]
Just some fixes to the comments, otherwise this looks good.  Thanks for helping to improve our testing coverage!

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 1 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/editable_text_test.dart:5054]: Fixed, I added a `pump()` above.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 2 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 3 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 4 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 5 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done.
Thanks for review quickly :)

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 6]
```diff
@@ -4998,6 +4998,89 @@ void main() {
...
+
+    state.updateEditingValue(const TextEditingValue(
+      text: 'a',
+    ));
+    await tester.pump();
+
+    // Nothing called when only the remote changes
+    expect(log.length, 0);
+
+    // Hide the keyboard
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5057]: Missing a period at the end.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 7]
```diff
@@ -4998,6 +4998,89 @@ void main() {
...
+    // Close the InputConnection.
+    expect(methodCall, isMethodCall('TextInput.clearClient', arguments: null),);
+    methodCall = log[1];
+    expect(methodCall, isMethodCall('TextInput.hide', arguments: null),);
+    // The keyboard loses focus.
+    expect(focusNode.hasFocus, false);
+
+    log.clear();
+
+    // Send repeat value from the engine
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5072]: Missing a period at the end.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 8]
```diff
@@ -4998,6 +4998,89 @@ void main() {
...
+    state.updateEditingValue(const TextEditingValue(
+      text: 'a',
+    ));
+    await tester.pump();
+
+    // Nothing called when only the remote changes
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5078]: Missing a period at the end.

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/test/widgets/editable_text_test.dart:5054]: This still needs a period at the end of the sentence:

`// Nothing called when only the remote changes.`

Sorry for the strict grammar enforcement!

**Review by justinmc — COMMENTED** [review]
Just some periods to fix quickly, then this is ready to merge.

**Comment by goderbauer** [comment]
(PR triage): @xu-baolin Do you have plans to come back to this soon to address the feedback? Thanks.

**Comment by xu-baolin** [comment]
> (PR triage): @xu-baolin Do you have plans to come back to this soon to address the feedback? Thanks.

I'm sorry I didn't reply in time. I'll finish my holiday tomorrow and get back to work right away.

**Comment by xu-baolin** [comment]
> Just some periods to fix quickly, then this is ready to merge.

Sorry for not understanding what you mean, it has been fixed now

**Review by justinmc — APPROVED** [review]
LGTM 👍 

There is a merge conflict though, can you resolve it?  Then this is good to merge.

**Comment by justinmc** [comment]
Thanks, it will be merged when the tree is green now!

**Pull request merged**